src/pages/user-descriptor/line-graph.jsx

```diff
@@ -1,59 +1,45 @@
 import React from "react";
 import ReactEcharts from "echarts-for-react";
 
-// Expected Object
-function BossObject(Unit, Weight, Latitude, Longitude, Time, Gender) {
-  this.unit = Unit;
-  this.weight = Weight;
-  this.latitude = Latitude;
-  this.longitude = Longitude;
-  this.time = Time;
-  this.gender = Gender;
+// GraphQL
+import { gql } from "apollo-boost";
+import { useQuery } from "@apollo/react-hooks";
+
+const lgData = gql`
+  {
+    getPaginatedDescriptors(unit: "lb", start: 0, count: 400) {
+      value
+      unixTimestamp
+    }
+  }
+`;
+
+function DataFunction() {
+  const { loading, error, data } = useQuery(lgData);
+
+  if (loading) return "Loading...";
+  if (error) return `Error! ${error.message}`;
+
+  return data.getPaginatedDescriptors;
 }
 
+function BossObject(Gender) {
+  this.gender = Gender;
+}
 // object array
 let dataArray = [];
-
 // Generated random weight and time.
 for (let i = 0; i < 100; i++) {
-  let Unit = "lb";
-  let Latitude = 0;
-  let Longitude = 0;
-  let startDate = new Date(2012, 0, 1);
-  let endDate = new Date();
-  let randomTime = new Date(
-    startDate.getTime() +
-      Math.random() * (endDate.getTime() - startDate.getTime())
-  );
-  let maxWeight = 350;
-  let minWeight = 40;
-  let Weight =
-    Math.floor(Math.random() * (maxWeight - minWeight + 1)) + minWeight; // newValue is the weight in this case.
-
-  // let date = randomTime.getDate();
-  let month = randomTime.getMonth(); //Be careful! January is 0 not 1
-  let year = randomTime.getFullYear();
   let randGen = Math.floor(Math.random() * 2);
-  let dateString = month + 1 + "-" + year;
   let newGender = "";
   if (randGen) {
     newGender = "Male";
   } else {
     newGender = "Female";
   }
-
-  let newWTobject = new BossObject(
-    Unit,
-    Weight,
-    Latitude,
-    Longitude,
-    dateString,
-    newGender
-  );
-
+  let newWTobject = new BossObject(newGender);
   dataArray.push(newWTobject);
 }
-
 let sortedTimeArray = [];
 for (let j = 0; j < dataArray.length; j++) {
   sortedTimeArray[j] = dataArray[j].time;
```

Comment:
What is this piece of code doing now? dataArray[j].time should be null since all objects in dataArray has only a gender property. If not needed, delete unneeded code.